### ISSUE TITLE
Fall 2020 security package update

### ISSUE THREAD

**Issue by th0ma7** [SynoCommunity/spksrc#4211]
Issue for tracking all packages needing publishing following major updates to `openssl`, `glib`, `gnutls` & `libgcrypt` adressing multiple security issues namely:

* https://www.cvedetails.com/vulnerability-list/vendor_id-217/product_id-383/version_id-194420/Openssl-Openssl-1.0.2g.html
* https://www.cvedetails.com/vulnerability-list/vendor_id-283/product_id-16275/version_id-260024/Gnome-Glib-2.56.1.html
* https://www.cvedetails.com/vulnerability-list/vendor_id-72/product_id-4433/GNU-Gnutls.html
* https://www.cvedetails.com/vulnerability-list/vendor_id-4711/product_id-25777/Gnupg-Libgcrypt.html

Related PR: 
* ~#4155: Update openssl~ (merged)
* ~#4010: Update glib, gnutls & libgcrypt~ (merged)
* ~#4209: Update ruTorrent~ (merged, see Package updates below)
* ~#4195: Update synocli-net~ (merged)
* ~#4218 Update to tvheadend~ (merged)
* ~Update mosquitto (todo: create new PR after merge of #4115)~

Indirectly related PR:
* ~#4200: Update to `make all-supported` with parallel build to ease publishing~ (merged)
* ~#4208: Remove bitlbee~ (merged)

### Package updates
- [ ] boxbackup-client        0.12-2
- [x] ffmpeg                    4.3.1
- [x] fossil-scm              2.12.1-2 (~BUG: error while loading shared libraries: libssl.so.1.1: cannot open shared object file~ fixed with #4241)
- [x] git                     2.28.0-16
- [x] gnupg                     2.2.23-3 (#4010, #4233)
- [ ] haproxy                 2.2.4-22 (#4010)
- [x] links                   2.21-2 (moved to synocli-net with #4195)
- [x] mediainfo          20.08-11 (consolidate with #4209)
- [x] ntopng          4.2-3 (consolidate with #4255 + download `libmysqlclient` package fix #4245 + libzmq duplicate removal #4254)
- [x] sslh            1.21c-5  (#4010 + fix with  #4246 + libcap #4248)
- [x] synocli-file            1.2-4  (update midnight commander to 4.8.25)
- [x] rutorrent                    3.10  (BUG: service does not start - consolidate with #4209 and #4234)
- [x] tvheadend             20201011-9ed76c0-26 (Oct 11th 2020 #4218)

#### First stage
Before or in parallel with updated packages above
- [x] python3
- [x] python

#### Second Stage
- [ ] borgbackup
- [ ] deluge
- [ ] domoticz
- [x] homeassistant
- [ ] icecast
- [ ] irssi
- [x] itools (included in synocli-net, no dedicated package anymore)
- [ ] lidarr
- [ ] mtproxy
- [x] mutt
- [x] ntopng
- [ ] nzbdrone (sonarr)
- [ ] radarr
- [x] synocli-disk
- [x] synocli-monitor (#4329)
- [x] synocli-net (consolidate with #4195 and #4010)
- [x] transmission
- [x] tvheadend
- [ ] umurmur
- [x] znc

### Packages with new revision due to `glib`,`gnutls` or `libgcrypt`  (PR #4010)
- [x] ffmpeg
- [x] chromaprint
- [x] comskip
- [x] gnupg
- [x] ~sshfs~ (part of `synocli-net`)
- [x] sslh (BUG: service does not start - fixed with  #4246)
- [ ] icecast
- [ ] irssi
- [x] ntopng
- [x] python
- [x] python3
- [x] rutorrent
- [x] synocli-disk (#4252)
- [x] synocli-file (#4242)
- [x] synocli-net (including #4257)
- [x] tvheadend

### Packages with new revision due to `rutorrent` update  (PR #4209)
- [x] mediainfo
- [x] rutorrent (BUG: service does not start)

### To be confirmed (issue #3666)
- [x] mono

### Packages set for removal  (PR #4208)
- [x] bitlbee

**Comment by th0ma7** [comment]
Proposal:
1. As @hgy59 is proposing to start with a first block for `python` and `python3` (with or without affiliated packages)?
2. I would suggest a second block including the classic 4x related to `ffmpeg` (e.g. `chromaprint`, `comskip` and `tvheadend`).
3. A pause for a week to confirm no major blockers emerge
4. publish the remaining

**Comment by hgy59** [comment]
>     1. Merge all PR starting with #4155 with rebasing for all subsequent to confirm proper github-action builds

Before this, we should merge #4191 to fix the Makefile of openssl 1.0.1 and then rebase #4155 before continuing with the proposed list.

**Comment by ymartin59** [comment]
So do I understand well it is worth waiting for glib/gnutls/libgcrypt merge before building and testing again python+python3?
From my point of view, there is no need to deliver Python related application packages.

**Comment by th0ma7** [comment]
Exactly as there are sub-dependencies that relies on this.  But everything is building beautifully currently and addressing the last remaining bits such as spk package rev & changelogs.

**Comment by hgy59** [comment]
@ymartin59 I published synocli-net to the repo about 20 hours ago. It is shown in the web page but in the package center of my diskstations still the old version 1.4-4 is shown. And yes, I have activated the packages. It there anything else that is missing or do we need to wait for a "cache update"?

**Comment by th0ma7** [comment]
I noticed from time to time that it can take as much as 48h+ to finally see the new packages...
Note that `synocli-net` is also indirectly affected by #4010 as well.  Although not convinced is worth pushing another update...

**Comment by ymartin59** [comment]
There is a chance both combination of Package Center refresh period and Fastly caching configuration delay availability in users' DSM

**Comment by ymartin59** [comment]
I think Mono also requires another build #3666

**Comment by ymartin59** [comment]
@th0ma7 When building fossil-scm, openssl PLIST libraries have not been included into package... No idea if it is a transient error when building, or if it is a trouble with build chain changes.

**Comment by th0ma7** [comment]
I've just check with the `tvheadend` package I built this morning and everything looks good?
```
~/git-master/spksrc/cross/openssl$ cat PLIST | while read LINE
> do
> grep $LINE ~/git-master/spksrc/spk/tvheadend/work-apollolake-6.2.3/PLIST
> done
rsc:bin/c_rehash
bin:bin/openssl
lib:lib/engines-1.1/afalg.so
lib:lib/engines-1.1/capi.so
lib:lib/engines-1.1/padlock.so
lnk:lib/libcrypto.so
lib:lib/libcrypto.so.1.1
lnk:lib/libssl.so
lib:lib/libssl.so.1.1
```

**Comment by th0ma7** [comment]
I just tried it against `fossil-scm` and indeed there is an issue out there:
```
$ more PLIST 
bin:bin/fossil
```
Dependency tree looks good (although with a duplicate `zlib`):
```
fossil-scm$ make dependency-tree
+ fossil-scm
	+ fossil-scm 2.12.1
		+ zlib 1.2.11
		+ openssl 1.1.1h
			+ zlib 1.2.11
```

**Comment by th0ma7** [comment]
I'll do a git-bisect and try to figure out where the issue is exactly... more to come with hopefully a fix shortly.

**Comment by th0ma7** [comment]
I need help from somebody to tell me that I am completely mistaking and crazy...
* `fossil-scm` package seems to trigger the problem, always.
* At the opposite `zsh` works out all right.

Findings so far the issue has been going since at least March 2019.
Actually I've stopped searching at that point just because I now doubt that my method is right.
A second opinion would be much appreciated.
```
commit ab347af3e91141c0125a67306dde97b9f73c01c6 (HEAD)
Author: hgy59 <hpgy59@gmail.com>
Date:   Sat Mar 30 09:06:52 2019 +0100
```

**Comment by th0ma7** [comment]
BTW @ymartin59 let me know how if you need some help to deploy other packages.
I might be able to find a few spare cycles over the week-end to help out.

**Comment by th0ma7** [comment]
Foud a few cycles to publish `mediainfo`, done.

**Comment by ymartin59** [comment]
@th0ma7 Do not hesitate to pick any package you are interested in. According to remaining list, most of them require "update" process to publish latest version.

**Comment by th0ma7** [comment]
@hgy59 was  links 2.21-2 (todo: move to synocli-net with #4195) completed par of the new release of `synocli-net` ?

**Comment by th0ma7** [comment]
@hgy59 thnx for releasing `ntop` and `lua`.  I was working on `haproxy` but work on this got lost on my failed hard drive before I was able to push my git upstream.

**Comment by BenjV** [comment]
There is a security alert for Openssl with a high risk classification reported.
see:
https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-3711
and
https://www.openssl.org/news/vulnerabilities.html

This is fixed in Openssl 1.1.1l.
Is it a good idea to upgrade to this openssl version?

**Comment by hgy59** [comment]
Closing this, as all remaining packages are covered by #4820.
